### PR TITLE
chore(update): add hint for JUEL method resolution

### PR DESCRIPTION
**Pull request by tmetzke**
related to https://github.com/camunda/camunda-bpm-platform/issues/3053

### CODE REVIEW

**Comment by tmetzke** [comment]
I am not entirely sure if this creates more uncertainty than necessary. Looking for feedback. 🙂

File: content/update/patch-level.md [hunk 1]
```diff
@@ -404,6 +404,18 @@ With this release, we adjusted the [Optimistic Locking]({{< ref "/user-guide/pro
...
+
+### Bean method resolution from JUEL expressions
+
+Calling bean methods from JUEL expressions now works more reliably and supports overloaded methods. As a result, the behavior of calling overloaded methods from such expressions changes.
+
+Previoulsy, overloaded methods couldn't be determined reliably and were chosen based on the order of the underlying JDK's result of the call to `Class#getMethods` for a specific method name. As a result, the expected method might have been chosen for the expression evaluation but this could not be guaranteed.
+
+Now, the most specific version according to the provided number and types of method parameters is chosen consistently. For example, method `myMethod` expecting an `Integer` is chosen over method `myMethod` expecting an `Object` if the provided parameter is an `Integer` or can be coerced into one.
+
+Ideally, you shouldn't notice any difference in method invocation from JUEL expressions. However, we recommend testing your existing expressions thoroughly before using version in production.
```
Comment (tasso94) [inline content/update/patch-level.md:417]:
```suggestion
Ideally, you shouldn't notice any difference in method invocation from JUEL expressions. However, we recommend testing your existing expressions thoroughly before using this patch version in production.
```

**Review by tasso94 — APPROVED** [review]
> I am not entirely sure if this creates more uncertainty than necessary.

I think making this transparent creates value:

* If the changed behavior breaks the tests of a user unexpectedly, they can easily understand that this might be because we changed the behavior slightly. 
* In the past I saw JDKs that behave differently then others. Maybe there exist JDKs that reliably returned the wrong method and users took that for granted. I don't expect this happens for the majority of users. However, those that struggle can learn on their own how to tweak their implementation instead of creating support cases which ideally means less support load.

**Comment by tmetzke** [comment]
> > I am not entirely sure if this creates more uncertainty than necessary.
> 
> I think making this transparent creates value:
> 
> * If the changed behavior breaks the tests of a user unexpectedly, they can easily understand that this might be because we changed the behavior slightly.
> * In the past I saw JDKs that behave differently then others. Maybe there exist JDKs that reliably returned the wrong method and users took that for granted. I don't expect this happens for the majority of users. However, those that struggle can learn on their own how to tweak their implementation instead of creating support cases which ideally means less support load.

Thanks, @tasso94, this is really helpful 👍

**Review by christinaausley — APPROVED** [review]
Nice 👍 No adjustments from me on this one.

**Review by yanavasileva — APPROVED** [review]

**Pull request merged**